src/backend/distributed/utils/resource_lock.c

```diff
@@ -1060,3 +1086,218 @@ CitusLockTableAclCheck(Oid relationId, LOCKMODE lockmode, Oid userId)
 
 	return aclResult;
 }
+
+
+/*
+ * AcquireDistributedLockOnRelations_Internal acquire a distributed lock on worker nodes
+ * for given list of relations ids. Worker node list is sorted so that the lock
+ * is acquired in the same order regardless of which node it was run on.
+ * Notice that no lock is acquired on coordinator node if the coordinator is not
+ * added to the metadata.
+ *
+ * Notice that no validation is done on the relationIds, that is the responsibility of
+ * AcquireDistributedLockOnRelations.
+ *
+ * A nowait flag is used to require the locks to be available immediately
+ * and if that is not the case, an error will be thrown
+ */
+static void
+AcquireDistributedLockOnRelations_Internal(List *relationIdList, LOCKMODE lockMode, bool
+										   nowait)
+{
+	Oid relationId = InvalidOid;
+	List *workerNodeList = ActivePrimaryNodeList(NoLock);
+	Oid userId = GetUserId();
+
+	const char *lockModeCString = LockModeToLockModeCString(lockMode);
+
+	/*
+	 * We want to acquire locks in the same order across the nodes.
+	 */
+	workerNodeList = SortList(workerNodeList, CompareWorkerNodes);
+
+	UseCoordinatedTransaction();
+
+	int32 localGroupId = GetLocalGroupId();
+
```

Comment:
If we send `LOCK TABLE` command, do we need to iterate on the relationId? In otherwords, can't we  / shouldn't we need to send `LOCK TABLE t1,t2,t3`;  in the same order that user types in? (or the relations we calculate for TRUNCATE CASCADE)